src/store/reducers/subjects.js

```diff
@@ -0,0 +1,57 @@
+import { ADD_SUBJECT, TOGGLE_SUBJECT, REMOVE_SUBJECT, UPDATE_SUBJECT } from '../../constants/ActionTypes'
+
+const subjects = (state = { primary: [], lingual: [] }, action) => {
+  switch (action.type) {
+    case ADD_SUBJECT:
+      return {
+        ...state,
+        [action.payload.subjectsType]: [
+          ...state[action.payload.subjectsType],
+          {
+            id: action.payload.id,
```

Comment:
Despite being saved under `action.payload.subjectsType` key in store, the subject should have property `type` set within itself, as this is checked in places like 'subjectPicker'